# Movimiento con Aceleración Constante

In [2]:
from IPython.display import HTML
from IPython.display import display

# Tomado de https://stackoverflow.com/questions/31517194/how-to-hide-one-specific-cell-input-or-output-in-ipython-notebook
tag = HTML('''<script>
code_show=false; 
function code_toggle() {
    if (code_show){
        $('div.cell.code_cell.rendered.selected div.input').hide();
    } else {
        $('div.cell.code_cell.rendered.selected div.input').show();
    }
    code_show = !code_show
} 

$( document ).ready(code_toggle);
</script>

Para mostrar/ocultar código presione <a href="javascript:code_toggle()">aquí</a>.''')



###########################################################################################################################


from matplotlib import animation
import numpy as np
import matplotlib.pyplot as plt
import ipywidgets as widgets
from IPython.display import display 
from ipywidgets import interactive



g  = -9.81
v = 5
h = 40

def posicion(t):
    #Ecuacuion de movimiento con aceleracion constante
    global g
    global v
    global h
    t = float(t - tiempoAdicional(v, h))
    y = v*t + (g/2)*t**2
    return y

def velocidad(t):
    global g
    global v
    global h
    t = float(t-tiempoAdicional(v, h))
    velocidad = v + g*t
    return velocidad

def tiempoAdicional(v, h):
    return h/v

t1 = 0.25 + tiempoAdicional(v, h)

t2 = 1 + tiempoAdicional(v, h)

t3 = 8 + tiempoAdicional(v, h)

deltaT = 0.05

def Inciso_A ():
    
    display (tag)

    def arregloPosicionesVelocidades(t):
        global v
        t = float(t)
        mru = v*t
        if(mru <= 40):  
            return [mru, 5]
        elif(mru >= 0):
            return [40 + posicion(t), velocidad(t)]
        else:
            return [0,0.3]

    tiempos1 = np.arange(0, t1+deltaT, deltaT)
    tiempos2 = np.arange(0, t2+deltaT, deltaT)
    tiempos3 = np.arange(0, t3+deltaT, deltaT)

    pos1 = widgets.Label (value="La posicion en $t_1$ es de: "
                                 + str(round(arregloPosicionesVelocidades(t1)[0],3))+ "$m$ sobre el suelo")

    vel1 = widgets.Label (value="La velocidad en $t_1$ es de: "
                                 + str(round(arregloPosicionesVelocidades(t1)[1],3))+ "$m/s$ sobre el suelo")

    pos2 = widgets.Label (value="La posicion en $t_2$ es de: "
                                 + str(round(arregloPosicionesVelocidades(t2)[0],3))+ "$m$ sobre el suelo")

    vel2 = widgets.Label (value="La velocidad en $t_2$ es de: "
                                 + str(round(arregloPosicionesVelocidades(t2)[1],3))+ "$m/s$ sobre el suelo")

    display (pos1, vel1, pos2, vel2)
    
    
###########################################################################################################################


def Inciso_B ():
    
    display (tag)
    
    import math


    discriminante = v**2 - 4 *  1/2 * g * h


    x1 = (-v + math.sqrt (discriminante)) / (2 * 1/2 * g)

    x2 = (-v - math.sqrt (discriminante)) / (2 * 1/2 * g)

    sol1 = widgets.Label (value="Una solución para t es: "
                                 + str(round(x1,3))+ "$s$")

    sol2 = widgets.Label (value="Una solución para t es: "
                                 + str(round(x2,3))+ "$s$ ")

    display (sol1, sol2)
    
    return discriminante, x2
     
          

###########################################################################################################################


def Inciso_C (tiempo):
    
    display (tag)
    
    velFinal = v + g * tiempo

    solVel = widgets.Label (value="La velocidad con la que llega al suelo es: "
                                 + str(round(velFinal,3))+ "$m/s$")

    display (solVel)
    

###########################################################################################################################


def Inciso_D (discriminante):
    
    display (tag)
    
    yMax = -discriminante / (4 * 1/2 * g)

    yMaxEtiqueta = widgets.Label (value="La altura máxima que alcanza el saco es de: "
                                 + str(round(yMax,3))+ "$m$")

    display (yMaxEtiqueta)
    
    
###########################################################################################################################


def Inciso_E ():
    
    display (tag)

    def CalculoVelocidad (arregloTiempo):
        velocidad = v + g * arregloTiempo
        return velocidad

    def CalculoPosición (arregloTiempo):
        posición = h + v * arregloTiempo + 1/2 * g * arregloTiempo ** 2
        return posición

    def GraficoInteractivo(tiempo):
         
        arregloTiempo = np.linspace (0, tiempo, 20)
        posiciónActual = CalculoPosición (arregloTiempo)
        velocidadActual = CalculoVelocidad (arregloTiempo)
        aceleraciónActual = np.ones (20) * g

        fig, (ax1, ax2, ax3) = plt.subplots (1, 3, figsize = (20,7), dpi= 120, sharex = True)

        ax1.set_xlabel('Tiempo $(s)$')
        ax1.set_ylabel('Posición $(m)$')
        ax1.plot (arregloTiempo, posiciónActual)
        ax1.set_xlim([0, 4]) 
        ax1.set_ylim([0, 50])  
        ax1.grid ()

        ax2.set_xlabel('Tiempo $(s)$')
        ax2.set_ylabel('Velocidad $(m/s)$')
        ax2.plot (arregloTiempo, velocidadActual) 
        ax2.set_xlim([0, 3.5])
        ax2.set_ylim([-40, 10]) 
        ax2.grid ()

        ax3.set_xlabel('Tiempo $(s)$')
        ax3.set_ylabel('Aceleración $(m/s^2)$')
        ax1.set_xlim([0, 3.5])
        ax3.set_ylim([-14, 8]) 
        ax3.plot (arregloTiempo, aceleraciónActual)
        ax3.grid ()
    
    
    play = widgets.Play(value=0, min=0, max=4, step=1, disabled=False)
    tieEtiqueta = widgets.Label (value="Tiempo $(s)$:")
    tieSlider = widgets.FloatSlider(min=0, max=4,)
    widgets.jslink((play, 'value'), (tieSlider, 'value'))
    cajaTiempo = widgets.HBox([tieEtiqueta, tieSlider])


    salida = widgets.interactive_output(GraficoInteractivo, {'tiempo':tieSlider})

    display(cajaTiempo, play, salida)

## Caída Libre

Un ejemplo claro de movimiento con aceleración constante es el caso de la caída libre. Esto debido a que el cuerpo está sometido, en situaciones ideales, solamente a la aceleración producida por la gravedad, la cual en la Tierra es de aproximadamente:
$$
a=|g|=9.81~\text{m/s²}
$$
Como ejemplo de este caso se analizará el siguiente problema.

El tripulante de un globo aerostático, que sube verticalmente con velocidad constante de magnitud $5.00~\text{m/s}$, suelta un saco de arena cuando el globo está a $40.0~\text{m}$ sobre el suelo, como se muestra en la siguiente figura.
<img src="Problema.png">
Después de que se suelta, el saco de arena está en caída libre. 

a) Calcule la posición y velocidad del saco a $0.250~\text{s}$ y $1.00~\text{s}$ después de soltarse.

b) ¿Cuántos segundos tardará el saco en chocar con el suelo después de soltarse? 

c) ¿Con qué velocidad chocará? 

d) ¿Qué altura máxima alcanza el saco en relación con el suelo? 

e) Dibuje las gráficas ay-t, vy-t y y-t para el movimiento.

Definimos el valor de la gravedad que usaremos para este ejercicio.
El sistema de coordenadas apuntará positivo a la derecha y arriba.
$$
a = g = -9.81~\text{m/s²}
$$

Se requiere el valor de la velocidad del globo que da el enunciado.
$$
v_0 = 5.00~\text{m/s}
$$

Se define según el enunciado el valor de la altura conseguida.
$$
h = 40~\text{m}
$$

## Solución

### a) Calcule la posición y velocidad del saco a $0.250~\text{s}$ y $1.00~\text{s}$ después de soltarse.
Para la primera parte del problema se definen las ecuaciónes necesarias. Al ser un desplazamiento en el eje $y$ se utiliza esta variable.

$$
y = y_0 + v_0t + \dfrac{1}{2}at^2
$$

$$
v = v_0 + at
$$

Para esta primera parte, simplemente se debe sustituir las variables por los datos brindados en el enunciado.

In [3]:

Inciso_A ()

Label(value='La posicion en $t_1$ es de: 40.943$m$ sobre el suelo')

Label(value='La velocidad en $t_1$ es de: 2.547$m/s$ sobre el suelo')

Label(value='La posicion en $t_2$ es de: 40.095$m$ sobre el suelo')

Label(value='La velocidad en $t_2$ es de: -4.81$m/s$ sobre el suelo')

### b) ¿Cuántos segundos tardará el saco en chocar con el suelo después de soltarse?
En el inciso b se solicita el tiempo que tarda en chocar al suelo. Por lo que se puede utilizar la ecuación:

$$
y = y_0 + v_0t + \dfrac{1}{2}at^2
$$

La cual es una ecuación cuadrática por lo que se pueden obtener 2 soluciones.

In [9]:
solucion = Inciso_B ()

discriminante = solucion [0]
tiempo = solucion [1]

Label(value='Una solución para t es: -2.391$s$')

Label(value='Una solución para t es: 3.41$s$ ')

La solución negativa se descarta porque no tiene sentido físico en esta situación. Se concluye que el saco chocó contra el suelo a los $3.41~\text{s}$ de ser lanzado

### c) ¿Con qué velocidad chocará?
En el caso del inciso c se solicita la velocidad final, por lo que se puede utilizar la siguiente ecuación.
$$
v = v_0 + at
$$

Para la cual solo se deben realizar las sutituciones de las variables por los valores dados y los obtenidos. Por lo que:


In [10]:
Inciso_C (tiempo)

Label(value='La velocidad con la que llega al suelo es: -28.457$m/s$')

### d) ¿Qué altura máxima alcanza el saco en relación con el suelo?
Para el inciso d se solicita la altura que alcanza el saco. Se utiliza la ecuación:
$$
y = y_0 + v_0t + \dfrac{1}{2}at^2
$$

Al ser cuadrática se sabe que el vértice en el eje $y$ es igual a:

$$
y_{max}= \dfrac{\sqrt{v^2-4\dfrac{g}{2}h}}{4\dfrac{g}{2}}
$$



In [4]:
Inciso_D (discriminante)

Label(value='La altura máxima que alcanza el saco es de: 41.274$m$')

### e) Dibuje las gráficas ay-t, vy-t y y-t para el movimiento.

Finalmente, graficando el movimiente  se obtiene:

In [11]:
Inciso_E ()

Play(value=0, max=4)

Output()